src/algorithms/digi/PhotoMultiplierHitDigi.cc

```diff
@@ -137,13 +140,41 @@ eicrecon::PhotoMultiplierHitDigi::AlgorithmProcess(std::vector<const edm4hep::Si
                   it.first,
                   static_cast<decltype(edm4eic::RawTrackerHitData::charge)>(data.signal),
                   static_cast<decltype(edm4eic::RawTrackerHitData::timeStamp)>(data.time/m_cfg.timeStep)
-                  //,pos2vec(data.pos) // TEST gap cuts; requires member `edm4hep::Vector3d position` in data model datatype
+                  //,pos2vec(data.pos_local) // TEST gap cuts; requires member `edm4hep::Vector3d position` in data model datatype
                 };
                 raw_hits.push_back(hit);
             }
         }
-        return raw_hits;
 
+        //build noise raw hits
+        if (m_cfg.enableNoise) {
+          m_log->trace("{:=^70}"," BEGIN NOISE INJECTION ");
+          std::unordered_map<uint64_t, std::vector<HitData>> hit_groups_noise;
+          float p = m_cfg.noiseRate*m_cfg.noiseTimeWindow*dd4hep::ns;
```

Comment:
We eventually want to put the units in `DRICH.cc` instead, that is,
```cpp
digi_cfg.noiseTimeWindow = 20.0 * dd4hep::ns; // [ns]  // DRICH.cc
```
and
```cpp
float p = m_cfg.noiseRate*m_cfg.noiseTimeWindow / dd4hep::ns;  // here
```
This is a more general issue though, and I plan to take care of all these config parameter units in a separate PR.

But my main question here is do you really need the `*dd4hep::ns` here?

Reply:
I moved it to DRICH.cc. It is needed to convert the time windows in seconds (the rate in Hz). But I don't think I need it when I estimate the probability.